src/main/java/uk/gov/justice/digital/delius/transformers/ConvictionTransformer.java

```diff
@@ -0,0 +1,98 @@
+import uk.gov.justice.digital.delius.data.api.Offence;
+import uk.gov.justice.digital.delius.data.api.Sentence;
+import uk.gov.justice.digital.delius.jpa.standard.entity.*;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+import static uk.gov.justice.digital.delius.transformers.TypesTransformer.zeroOneToBoolean;
+
+@Component
+public class ConvictionTransformer {
+    private final MainOffenceTransformer mainOffenceTransformer;
+    private final AdditionalOffenceTransformer additionalOffenceTransformer;
+
+    public ConvictionTransformer(MainOffenceTransformer mainOffenceTransformer, AdditionalOffenceTransformer additionalOffenceTransformer) {
+        this.mainOffenceTransformer = mainOffenceTransformer;
+        this.additionalOffenceTransformer = additionalOffenceTransformer;
+    }
+
+    public Conviction convictionOf(Event event) {
+        return Conviction.builder()
+                .active(zeroOneToBoolean(event.getActiveFlag()))
+                .convictionDate(event.getConvictionDate())
+                .referralDate(event.getReferralDate())
+                .convictionId(event.getEventId())
+                .index(indexOf(event.getEventNumber()))
+                .offences(offencesOf(event))
+                .sentence(Optional.ofNullable(event.getDisposal()).map(this::sentenceOf).orElse(null))
+                .inBreach(zeroOneToBoolean(event.getInBreach()))
+                .latestCourtAppearanceOutcome(Optional.ofNullable(event.getCourtAppearances()).map(this::outcomeOf).orElse(null))
+                .build();
+    }
+
+    private Long indexOf(String eventNumber) {
+        // no DB constraint so protected ourselves
+        return Optional.ofNullable(eventNumber)
+                .filter(StringUtils::hasText)
+                .filter(this::isNumber)
+                .map(text -> NumberUtils.parseNumber(text, Long.class))
+                .orElse(null);
+    }
+
+    private boolean isNumber(String text) {
+        return text.matches("\\d*");
+    }
+
+    private KeyValue outcomeOf(List<CourtAppearance> courtAppearances) {
+        return courtAppearances
+                .stream()
+                .filter(courtAppearance -> courtAppearance.getOutcome() != null).max(Comparator.comparing(CourtAppearance::getAppearanceDate))
+                .map(courtAppearance -> outcomeOf(courtAppearance.getOutcome()))
+                .orElse(null);
+    }
+    private KeyValue outcomeOf(StandardReference outcome) {
+        return KeyValue
+                .builder()
+                .description(outcome.getCodeDescription())
+                .code(outcome.getCodeValue())
+                .build();
+    }
+
+
+    private List<Offence> offencesOf(Event event) {
+        return ImmutableList.<Offence>builder()
+                .addAll(Optional.ofNullable(event.getMainOffence()).map(mainOffence -> ImmutableList.of(mainOffenceTransformer.offenceOf(mainOffence))).orElse(ImmutableList.of()) )
```

Comment:
Database would in theory allow an event without an offence, but in practice UI makes this impossible - either way I protect myself from bad data with the Null check